docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,105 @@
+cross-clusters connections can be based on requester IP addresses, which is not always easy to discover. 
+Therefore, later we will provide other isolation methods as an extension to NP.
+
+### Advantages:
+- does not require any changes in modules implementations, can be implemented in the Fybrik control plane only.
+- can be combined with other isolation methods
+
+### Disadvantages:
+- complicated in cross clusters deployments.
+- NP are implemented by the network plugin. Creating a NetworkPolicy resource without a controller - a Container Network 
+Interface (CNI) that implements it will have no effect. Default Kind deployment which by default uses "kindnetd” doesn’t 
+support NP. It requires installation another CNI, e.g. [Calico](https://github.com/projectcalico/calico). Here is the 
+[instructions](https://alexbrand.dev/post/creating-a-kind-cluster-with-calico-networking/). All K8s clusters in 
+production deployments use CNIs which support NP.
+- requires information about workload deployment, and/or setting labels to the workload
+
+## Implementation
+
+The “from” element of ingress 
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming network traffic based on “podSelectors” (labels based), and/or 
+“namespaceSelectors” (labels based) and/or “ipBlocks”. The last one permits incoming network traffic from predefined IP 
+address blocks.
+
+Current “FybrikApplicationSpec” has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of “clusterName” and “workloadSelector”. Unfortunately, based on this information we can 
+implement only one label based “podSelector”. Furthermore, for the backward compatibility of the current implementation 
+assumes that a Read scenario should have at least one label, which is not always true. 
+
+### Suggetions:
+
+- extend the “FybrikApplicationSpec.selector” element with an array of 
+[NetworkPolicyPeers](https://github.com/kubernetes/api/blob/59fcd23597fd090dba6b7e903eb0a8c9e8efb0a6/networking/v1/types.go#L183)
+this “NetworkPoliciesPeers” array will be an input element for the NP `from`element. This will restrict possible incoming
+connections to the relevant Fybrik module.
+- a blueprint will provide the ingress input data for chain connected modules.
+
+- deprecate the “workloadSelector” element, but continue to support it for backward compatibility. Process it as an 
+additional “podSelector” entry in the “NetworkPoliciesPeers” array
+
+- terminate the backward compatibility assumption that all Read flow should have at least a single workload label. 
+It was deprecated several Fybrik versions ago.
+
+- change the Flow field to be required instead of optional.
+
+- if the NetworkPolicyPeers array and “workloadSelector” are empty, no Network Policies will be created, which is 
+similar to current implementation.
```

Comment:
So no isolation in non-read/write flows?

Reply:
Examples ? Delete module doesn't have a server, no one can connect to it. Data movers are deployed by Fybrik, without input from a FybrikApplication, and blueprint knows all required information about theire sources and destinations.

Reply:
The modules deployed even in ingest/copy/delete flows (which have no workload) should have isolation shouldn't they?  I.e. A copy, quality, delete, etc module shouldn't be able to access data stores, other modules, etc except what is described in the plotter flow.

Reply:
Yes, it is described in the Plotter/blueprint flow, user should not and cannot provide this information. The bullets talks about the Workload<->data Plane connection (the first module) only.